docs/adr/013-add-provisioning-saga.md

```diff
@@ -0,0 +1,72 @@
+# 13. Add Provisioning Saga Workflow
+
+Authors: Adi Abramovich
+
+## Status
+
+Proposed
+
+## Problem Statement
+
+This ADR exists because there is a need of making the build and launch workflow more user-friendly. 
+this ADR will include three propositions regarding how should we implement this feature and what should we take into consideration.
+
+## Goals
+
+* Make the saga workflow fast, secure, and simple
+* Choose one of the architectures prop3os3ed
+
+## Non-goals
+
+## Current Architecture
+
+* Currently the Image builder wizard and the Provisioning Wizard are not connected.
+After collecting all the information needed in order to create an image The image builder wizard closes and we are redireced to Image builder main page which lists all the images that already have been created.
+Only after the image was created successfully, a **"Launch"** button appears which leads to the Provisioning wizard.
+
+## Proposed Architecture
+
+I have thought about three architecture options where all three options have the first steps in common:
+
+* Image builder collects all the information needed to create an image on a specific hyper-scaler
+* Pressing the **Build and launch** button on the wizard
+* Validate the build manifest
+* A background job is running in order to create an image
+* The Provisioning wizard opens up
+
+Important information to remember:
+
+* The Provisioning is dependent on the image id in order to continue and finish the launch
+* The shared sources are given directly from the image variable from IB Wizard
+
+**In the following section I will elaborate on the differences:**
+
+1. Combining Image Builder wizard with the Provisioning wizard and showing the status of the instance on Image Builder main page:
+   * Add a pooling mechanism questioning Image Builder if the image is ready.
+   * The customer fills in the data needed in order to launch an image.
+   * The wizard closes, the pooling continues, and when it is done a job is enqueued with the image id.
+   * The status of the job is presented on Image Builder main page. (like the status is being presented in the wizard now)
+  
+2. Combining Image Builder wizard with the provisioning wizard and adding a step in the Provisioning wizard creating the image:
+   * The customer fills in the data needed in order to launch an image.
+   * We are adding a step in the Provisioning wizard showing the status of the image creation.
+   * After the image has been created the Provisioning workflow continues(creating a reservation, uploading pubkeys when needed and launching the image)
+  
+3. Combining Image Builder wizard with the provisioning wizard and showing the status of the instance on Inventory main page:
+   * Add a pooling mechanism questioning Image Builder if the image is ready.
```

Comment:
ditto

Reply:
ditto regarding what?

Reply:
Where does the polling runs?

Reply:
We are querying IB endpoint from the UI

Reply:
Here I was not making proposals (I don't love polling from UI, as closing a browser stops polling, which is not great).
Here I was asking if we could be clear on that in the text, so future readers know what were the proposals.